File: mix/benchmark/fib.ml

let rec fib = fun n -> 
  if n<2
  then n
  else fib (n-1) + fib (n-2)
  ;;

for i = 0 to 34 do
  Printf.printf "n=%d => %d\n" i (fib i)
done
;;

File: mix/benchmark/fib.pl
use strict;
sub fib{
  if($_[0] < 2){
    return $_[0];
  }
  else{
    return fib($_[0] - 1) + fib($_[0] - 2);
  }
}

for(my $i=0; $i<35; ++$i){
  print "n=$i => ".fib($i)."\n"
}
